### DISCUSSION TITLE
STM32Cube CLT

### DISCUSSION THREAD

**Discussion by multiplemonomials** [mbed-ce/mbed-os#225] [Ideas]
Check it out!  STMicro released a new standalone command line tools package! https://www.st.com/en/development-tools/stm32cubeclt.html.  This is quite cool as it means that users will no longer need to install STM32Cube IDE in order to be able to use the official STM32 debugger! 

In order to make it work, someone will have to update the FindSTM32CUBE script in order to handle the directory layout and install location of this new package.

**Comment by JohnK1987** [comment]
Ok, but there are few options:
1. In STM32CUBE we replace current STM32CubeIDE with STM32CubeCLT
2. Let both separately as upload methods STM32CUBEIDE and STM32CUBECLT
3. Let both together under STM32CUBE but use one what will be available but STM32CubeCLT preferred.

Because in case of developing something with HALs you probably need to have IDE installed anyway and then CLT is not necessary to use.

So?

**Reply by multiplemonomials** [reply]
I think option 3 is the way to go

**Reply by JohnK1987** [reply]
Then #228
